packages/extension-base/src/background/RequestExtrinsicSign.ts

```diff
@@ -15,9 +15,6 @@ export default class RequestExtrinsicSign implements RequestSign {
   }
 
   sign (registry: TypeRegistry, pair: KeyringPair): { signature: string } {
-    // inject the current signed extensions for encoding
```

Comment:
Not sure why this was removed? (As commented the signed extensions from the payload should be as-defined for the transaction)

Reply:
It removed from here, and used in a level above this... i.e while handling signingApprovedPassword so the registry that's is passed will always have current signed extensions injected..
https://github.com/KarishmaBothara/extension/blob/dfdc1ed1eba183d9477d5c2c61ac63a707ff2a4f/packages/extension-base/src/background/handlers/Extension.ts#L343